draft-ietf-taps-transport-security.md

```diff
@@ -844,6 +855,70 @@ and the per-message nonce in the clear. Everything else is encrypted.
+OpenVPN facilitates authentication using either a pre-shared static key or
+using X.509 certificates and TLS. In pre-shared key mode, OpenVPN derives
+keys for encryption and authentication directly from one or multiple static
+keys. In TLS mode, OpenVPN encapsulates a TLS handshake, in which both peers
+must present a certificate for authentication. After the handshake, both sides
+contribute random source material to derive keys for encryption and
+authentication using the TLS pseudo random function (PRF). OpenVPN provides the
+possibility to authenticate and encrypt the TLS handshake itself using a
+pre-shared key or passphrase. Furthermore, it supports rekeying using TLS.
+
+After authentication and key exchange, OpenVPN encrypts payload data, i.e., IP
+packets or Ethernet frames, and signs the payload using an HMAC function. The
+default cipher is BlowFish and the default message digest algorithm is SHA1,
+but an application can select an arbitrary cipher, key size, and message digest
+algorithm for the HMAC. OpenVPN peers may also support cipher negotiation
+(NCP). If both hosts allow, they can upgrade the cipher to AES-256-GCM.
+Blocks are padded to an even multiple of block size.
+
+OpenVPN can run over TCP or UDP. When running over UDP, OpenVPN provides a
+simple reliability layer for control packets such as the TLS handshake and key
+exchange. It assigns sequence numbers to packets, acknowledges packets it
+receives, and retransmits packets it deems lost. This reliability layer is not
+used for data packets, which prevents the problem of two reliability mechanisms
+being encapsulated within each other. When running over TCP, OpenVPN includes
+the packet length in the header, which allows the peer to deframe the TCP
+stream into messages.
+
+For replay protection, OpenVPN assigns an identifier to each outgoing packet,
+which is unique for the packet and the currently used key. In pre-shared key
+mode or with a CFB or OFB mode cipher, OpenVPN combines a timestamp with an
+incrementing sequence number into a 64-bit identifier. In TLS mode with CBC
+cipher mode, OpenVPN omits the timestamp, so identifiers are only 32-bit. This
+is sufficient since OpenVPN can guarantee the uniqueness of this identifier for
+each key, as it can trigger rekeying if needed.
+
+OpenVPN supports connection mobility by allowing a peer to change its IP
+address during an ongoing session. When configured accordingly, a host will
+accept packets for a session from any IP address, as long as the packets are
+correctly authenticated.
+
+
+### Protocol Features
+
+- Peer authentication using certificates or pre-shared key
+- Mandatory mutual authentication.
+- Connection mobility
+- Out-of-order record receipt
+- Length-hiding padding
+
+See also the properties of TLS.
+
+### Protocol Dependencies
+
+- For control packets such as handshake and key exchange: Reliable transport. Reliability is provided either by TCP, or by OpenVPN's own reliability layer when using UDP.
```

Comment:
Should we add similar text to QUIC and DTLS?

Reply:
I think "Protocol dependencies" is a good section to say whether a protocol depends on transport features such as reliable data transfer. I guess we could add a sentence to both QUIC and DTLS protocol dependencies saying that the protocol does not depend on reliability, since it provides its own.
I'm not sure we need to explicitly say which protocol features a protocol does NOT need, but reliability is a pretty important feature, so maybe here it makes sense.
On the other hand, "relies on UDP" already implies that it doesn't need reliability?

Going through the other sections, I noticed that most other protocol state or imply whether they use TCP or UDP in their Protocol Dependencies (TLS, DTLS implies it, QUIC, IKEv2 implies it, tcpcrypt, Wireguard says "datagram", and CurveCP), only two do not include this information here (Secure RTP and MinimalT). Maybe we want to be consistent here.

Reply:
Yeah, consistency is the goal. Can you please take a stab at unifying those bits?

Reply:
Some protocols might be restricted to UDP. For the others, we should possibly say what guarantees are needed?e.g., many protocols work fine over UDP, and don't need reliability or ordering, but can equally well work on TCP when reliability and ordering are provided.

Reply:
Taking a stab, but I expect that we'll have to take a look at consistency again at some point (in some other pull request?).

@csperkins For example, RTP can run over UDP or TCP, right? Feel free to correct and/or give details.

Reply:
@theri RTP needs framing, but not reliability or ordering. It runs over TCP with an added (trivial) framing layer. 

Not sure what you want me to comment on, but I was thinking that rather than "runs on TCP" we'd say "requires a reliable, ordered, transport; has a binding to TCP", or similar?